certification/internal/policy/container/has_prohibited_packages.go

```diff
@@ -0,0 +1,114 @@
+package container
+
+import (
+	"path/filepath"
+	"strings"
+
+	rpmdb "github.com/knqyf263/go-rpmdb/pkg"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+// HasProhibitedPackages evaluates that the image does not contain prohibited packages,
+// which refers to packages that are not redistributable without an appropriate license.
+type HasNoProhibitedPackagesCheck struct{}
+
+func (p *HasNoProhibitedPackagesCheck) Validate(imgRef certification.ImageReference) (bool, error) {
+	pkgList, err := p.getDataToValidate(imgRef.ImageFSPath)
+	if err != nil {
+		log.Error("unable to get a list of all packages in the image")
+		return false, err
+	}
+
+	return p.validate(pkgList)
+}
+
+func (p *HasNoProhibitedPackagesCheck) getDataToValidate(dir string) ([]string, error) {
+	db, err := rpmdb.Open(filepath.Join(dir, "var", "lib", "rpm", "Packages"))
+	if err != nil {
+		return nil, err
+	}
+	pkgList, err := db.ListPackages()
+	if err != nil {
+		return nil, err
+	}
+	pkgs := make([]string, len(pkgList))
+	for i, pkg := range pkgList {
+		pkgs[i] = pkg.Name
+	}
+	return pkgs, nil
+}
+
+func (p *HasNoProhibitedPackagesCheck) validate(pkgList []string) (bool, error) {
+	var prohibitedPackages []string
+	for _, pkg := range pkgList {
+		_, ok := prohibitedPackageList[pkg]
+		if ok {
+			prohibitedPackages = append(prohibitedPackages, pkg)
+			continue
+		}
+		for _, name := range prohibitedPackageGlobList {
+			if strings.HasPrefix(pkg, name) {
+				prohibitedPackages = append(prohibitedPackages, pkg)
+				continue
+			}
+		}
+	}
+
+	log.Warn("The number of prohibited package found in the container image: ", len(prohibitedPackages))
+	if len(prohibitedPackages) == 0 {
+		return true, nil
+	}
+
+	log.Warn("found the following prohibited packages: ", prohibitedPackages)
+	return false, nil
```

Comment:
I'd probably rewrite this as follow to be consistent with the required label check:
```
if len(prohibitedPackages) > 0 {
    log.Warn("The number of prohibited package found in the container image: ", len(prohibitedPackages))
    log.Warn("found the following prohibited packages: ", prohibitedPackages)
}
return len(prohibitedPackages) == 0, nil
```